src/main/java/io/strimzi/kproxy/interceptor/RequestHandler.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.strimzi.kproxy.interceptor;
+
+import io.strimzi.kproxy.codec.DecodedRequestFrame;
+
+/**
+ * Handler for one or more Kafka protocol request types.
+ */
+public interface RequestHandler {
+
+    DecodedRequestFrame handleRequest(DecodedRequestFrame requestFrame);
```

Comment:
Yeah, this crossed my mind too. Certainly an option worth discussing, I'll log it as a follow-up. The one thing speaking against it would be that a given handler can only support exactly one message type. Not sure though whether this would be a real limitation in practice. We can decide once having agreed on the general path forward.

Reply:
>  a given handler can only support exactly one message type

Not as I imagined it. Note the method name (`handleProduceRequest` etc) above. So an Interceptor could multiply-inherit a number of `*RequestHandler` interfaces without a method name collision. The cost being the need for the switch in the `RequestHandlerAdapter.channelRead` as shown above.

Reply:
Ok, gotcha. Captured this in https://github.com/kroxylicous/kproxy/issues/8 for further exploration as a follow-up.